src/routes/new-reservation/+page.svelte

```diff
@@ -0,0 +1,448 @@
+    };
+
+    const dateOptions = {
+        year: "numeric",
+        month: "2-digit",
+        day: "2-digit",
+        hour: "numeric",
+        minute: "numeric",
+        hour12: false,
+    };
+
+    // Variables
+    let selectedStationIndex = 0;
+
+    // Global temporary save variables
+    let selectedStation;
+    let selectedBeginn;
+    let selectedEnd;
+    let selectedCar;
+
+    // Indicators
+    let beginIndicator = indicatorStatus.none;
+    let endIndicator = indicatorStatus.none;
+    let periodIndicator = false;
+    let emptyIndicator = false;
+
+    // Binding references
+    let beginReference;
+    let endReference;
+
+    // Formpage index for changing form view according to user reservation progress
+    let formPage = 0;
+
+    // Initialize promises as empty on mount
+    let cars = new Promise((resolve, reject) => {});
+    let stations = new Promise((resolve, reject) => {});
+
+    function resetIndicators() {
+        periodIndicator = false;
+        emptyIndicator = false;
+        beginIndicator = indicatorStatus.none;
+        endIndicator = indicatorStatus.none;
+    }
+
+    function fetchAvailableAllCars(loadedStations) {
+        cars = new Promise(async (resolve, reject) => {
+            try {
+                const searchParams = {
+                    stationId: loadedStations[selectedStationIndex].id,
+                    begin: new Date(beginReference.value).toISOString(),
+                    end: new Date(endReference.value).toISOString(),
+                };
+                const response = await tryFetchingRestricted(
+                    urls.get.availableCars +
+                        "?" +
+                        new URLSearchParams(searchParams).toString(),
+                );
+                if (response.ok) {
+                    resolve(await response.json());
+                } else {
+                    throw new Error("No car data found");
+                }
+            } catch (err) {
+                reject(err);
+            }
+        });
+    }
+
+    async function fetchAllStations() {
+        stations = new Promise(async (resolve, reject) => {
+            try {
+                const response = await tryFetchingRestricted(
+                    urls.get.allStations,
+                );
+                if (response.ok) {
+                    stations = await response.json();
+                } else {
+                    throw new Error("No stations found");
+                }
+            } catch (err) {
+                reject(err);
+            }
+        });
+    }
+
+    // Checks wether a car was selected and saves it
+    function checkAndSaveCarSelected(cars) {
+        const buttons = document.getElementsByName("radio-button");
+        for (let index = 0; index < buttons.length; index++) {
+            if (buttons[index].checked) {
+                selectedCar = cars[index];
+                return true;
+            }
+        }
```

Comment:
Das Auto würde ich unten beim onClick speichern. Hier dann den loop mit `forEach()`
```suggestion
    function checkCarSelected() {
        let checked = false;
        document.getElementsByName("radio-button").forEach(button => {
            if (button.checked) {
                checked = true;
            }
        });

        return checked;
    }
```

Reply:
und wo/wo speicher ich mir dann das Auto ab? Also woher weiß ich dann welches Auto ausgewählt ist?

Reply:
Ahh hab' vergessen die suggestion beim onClick zu machen. Hab das nur lokal erst bei mir getestet 😅

Reply:
Haben wir zusammen im Call angepasst und wird deshalb nicht mehr gebraucht, merci 🤩🤖